### ISSUE TITLE
Attempt to apply a sysctl attribute that does not exist on the system returns an error. Used to not. 

### ISSUE THREAD

**Issue by isuftin** [sous-chefs/sysctl#98]
### Cookbook version
0.10.1

### Chef-client version
12.21.4

### Platform Details
CentOS 6 in AWS

### Scenario:
Attempting to apply a sysctl attribute that does not exist on the system and hitting an error

### Steps to Reproduce:
Add a sysctl attibute ` 'sysctl' : { 'params' : { 'random.attribute=1' }}

### Expected Result:
If attribute does not exist within sysctl, do not attempt to apply it or attempt to apply it without erring out (send a warning to the logs)

### Actual Result:
In my case...

```
* sysctl_param[net.ipv6.conf.all.accept_ra] action apply

    ================================================================================
    Error executing action `apply` on resource 'sysctl_param[net.ipv6.conf.all.accept_ra]'
    ================================================================================

    Mixlib::ShellOut::ShellCommandFailed
    ------------------------------------
    Expected process to exit with [0], but received '255'
    ---- Begin output of sysctl -w "net.ipv6.conf.all.accept_ra=0" ----
    STDOUT:
    STDERR: error: "net.ipv6.conf.all.accept_ra" is an unknown key
    ---- End output of sysctl -w "net.ipv6.conf.all.accept_ra=0" ----
    Ran sysctl -w "net.ipv6.conf.all.accept_ra=0" returned 255

    Cookbook Trace:
    ---------------
    /var/chef/cache/cookbooks/sysctl/libraries/helpers_param.rb:62:in `set_sysctl_param'
    /var/chef/cache/cookbooks/sysctl/libraries/sysctl_param.rb:68:in `block (2 levels) in <class:SysctlParam>'
    /var/chef/cache/cookbooks/sysctl/libraries/sysctl_param.rb:64:in `block in <class:SysctlParam>'

    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/sysctl/recipes/default.rb

     26:   sysctl_param k do
     27:     value v
     28:   end
     29: end if node.attribute?('sysctl') && node['sysctl'].attribute?('params')

    Compiled Resource:
    ------------------
    # Declared in /var/chef/cache/cookbooks/sysctl/recipes/default.rb:26:in `block in from_file'

    sysctl_param("net.ipv6.conf.all.accept_ra") do
      action [:apply]
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      declared_type :sysctl_param
      cookbook_name "sysctl"
      recipe_name "default"
      value "0"
      key "net.ipv6.conf.all.accept_ra"
    end

    System Info:
    ------------
    chef_version=12.21.4
    platform=centos
    platform_version=6.9
    ruby=ruby 2.3.4p301 (2017-03-30 revision 58214) [x86_64-linux]
    program_name=chef-client worker: ppid=4884;start=16:23:13;
    executable=/opt/chef/bin/chef-client


Running handlers:
  - SyslogReporting
  - Chef::Handler::JsonFile
Running handlers complete
```

I recognize that `net.ipv6.conf.all.accept_ra` is valid, but I do not have it on my system and the attribute is coming in from an upstream cookbook that I am unable to remove via env/role attributes. Using version 0.9.0 of this cookbook seems to work fine in this regard.

**Comment by jiokmiso** [comment]
Hi, I made PR about adding -e flag control for sysctl command, and it is merged to master already. Once this gets released, you can turn on the flag `node['sysctl']['ignore_error']` to true, and you won't get a error from the sysctl command.

**Comment by damacus** [comment]
Closed by #99
